README.md

```diff
@@ -60,7 +60,14 @@ Once you have implemented a lower bound search for the start of the range, imple
 
 *How do you use binary search to find the lower bound of a number? How did you have to modify the binary search algorithm?*
 
+-When we use binary search to determine the lower bound, we return the idx of the smallest element equal to or larger than v instead of the idx of the element equal to v.
+We find this by searching in the upper half of x if x[mid]<v and by searching in the lower half if x[mid]>v. This is the same as in the ordinary binary search. In contrast to binary search, we continue to search in the lower half if x[mid]=v, to see if there is an element at a lower idx equal to v, and we do not stop searching until low = high.
+
 *Would anything be more difficult if the features covered ranges instead of single nucleotides (like real BED files)? What could go wrong, if anything?*
 
+- We would also have to comepare the end values for all the features with the upper value as one might start within the interval but end beyond the upper value. 
+
 *We wrote a tool for merging two BED files, but what if we had a bunch of them? What would the complexity be if we merged them in, one at a time? What would the complexity be if we merged all of the files at the same time?*
 
+- The complexity would be the same as with 2 BED files so O(n+m) (linear time) because we go trough both files from one end to the other. The complexity if we merge all of the BED files at the same time is also linear time and the length of all the files are added together so O(a+b+c+d+e). This is because we just compare all the values and then pick the smallest and add it to a new list, and then go to the nex value from the list it came from.   
```

Comment:
This is, unfortunately, a little too optimistic. If we merged two files of size $m$ $n$ times, it would indeed take $O(nm)$ (it takes time $2m$ to merge, and we do it $n$ times). But that is not the situation if we merge the files one at a time because we have a growing file that we merge into, and each time we add another file, the next merge will be more expensive.

For simplicity, let’s say that all the files have length $m$ and that there are $n$ of them. Merging the first two would take time $2m$ (two files of length $m$). Merging in the third would take $3m$, however (one file of length $2m$ and one of length $m$). The fourth would require time $4m$ (merging one file of length $3m$ with one of length $m$).

In general, we would need the time $m(2+3+4+\cdots+n-1)$ time to merge $n$ files of length $m$ and the sum in that expression (as you know) is in $O(n^2)$, so the running time is $O(mn^2)$ (far from linear).

Concatenating all the files and sorting (with something like merge sort, as you suggest) would instead take $O(nm\log(nm))$, which could be better (depending on $n$ and $m$).

You could also, using divide and conquer, split the input in two $n/2$-length lists of files, merge these recursively, and then merge the results. That would give you the recurrence $T(n,m) = 2T(n/2,m) + O(n\cdot m)$ which is in $O(mn\log(n))$.

If you merge the $n$ files in parallel, so you, in each iteration, find the smallest front element in the $n$ files, you need to copy $nm$ elements, but it takes time to find the smallest. If you do this with a linear scan, it takes $O(n)$ to find the smallest element in each iteration, and you end up back at $O(mn^2)$. You can speed this up (once we have seen heaps in a later lecture) so it takes time $O(\log n)$ to get the smallest element per iteration, and then you are back to $O(mn\log n)$ (just with a completely different approach).

Many roads lead to Rome.